plugin/trino-hive/src/test/java/io/trino/plugin/hive/BaseHiveConnectorTest.java

```diff
@@ -4949,6 +4946,45 @@ Actual rows (up to 100 of 1 extra rows shown, 1 rows in total):
                             [1, 2]""");
     }
 
+    @Test
+    public void testUnsupportedSetPartitionColumnType()
+    {
+        try (TestTable table = new TestTable(
+                getQueryRunner()::execute,
+                "test_unsupported_set_partition_column_type_",
+                "WITH(partitioned_by = ARRAY['part']) AS SELECT 1 AS id, 2 AS part")) {
+            assertQueryFails(
+                    "ALTER TABLE " + table.getName() + " ALTER COLUMN part SET DATA TYPE bigint",
+                    "Changing partition column types is not supported");
+            assertQuery("SELECT * FROM " + table.getName(), "VALUES (1, 2)");
+        }
+    }
+
+    @Override
+    protected void verifySetColumnTypeFailurePermissible(Throwable e)
+    {
+        assertThat(e).hasMessageContaining("Cannot change type");
+    }
+
+    @Override
+    protected Optional<SetColumnTypeSetup> filterSetColumnTypesDataProvider(SetColumnTypeSetup setup)
+    {
+        switch ("%s -> %s".formatted(setup.sourceColumnType(), setup.newColumnType())) {
+            case "bigint -> integer":
+            case "decimal(5,3) -> decimal(5,2)":
+                // Changing to small types is disallowed in the connector because it may lead to query failures
+                return Optional.of(setup.asUnsupported());
+            case "char(20) -> varchar":
+                // Hive connector trims trailing spaces after converting
+                return Optional.of(setup.withNewValueLiteral("rtrim(%s)".formatted(setup.newValueLiteral())));
+            case "timestamp(3) -> timestamp(6)":
+            case "timestamp(6) -> timestamp(3)":
+                // These precisions aren't supported in the configuration
+                return Optional.empty();
```

Comment:
`Optional.empty()` indicates the test cases totally skipped, i.e. something not working properly?

what happens if you do `asUnsupported`?

Reply:
It throws "Incorrect timestamp precision for timestamp(6); the configured precision is MILLISECONDS" when creating a table.